### PR TITLE
Qualification tool: Update speedup factor for few operators

### PR DESCRIPTION
**Pull request by nartal1**
This PR is to update speedup factor for some of the Execs. NDS logs were used to analyze the speed up of the operators between CPU and GPU logs.

Speedup of these execs are updated:
     BroadcastHashJoinExec
      ExchangeExec
      FilterExec
      HashAggregateExec
      SortExec
      SortMergeJoinExec

Changed the speedUp factor from Int to Double as the speedUp of some of the above exec is in Double. 
Updated the tests and rest of the code(minor update) accordingly.

### CODE REVIEW

**Comment by nartal1** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala [hunk 1]
```diff
@@ -2213,13 +2213,27 @@ object SupportedOpsForTools {
   private def operatorMappingWithScore(): Unit = {
     val header = Seq("CPUOperator", "Score")
     println(header.mkString(","))
+    val operatorCustomSpeedUp =  Map(
+      ("BatchScanExec", "2.0"),
+      ("BroadcastHashJoinExec", "3.0"),
+      ("ExchangeExec", "3.1"),
```
[the remark this answers is not in the document]
Comment (nartal1) [reply sql-plugin/src/main/scala/com/nvidia/spark/rapids/TypeChecks.scala]: done

File: tools/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala [hunk 2]
```diff
@@ -282,11 +282,12 @@ class SQLPlanParserSuite extends FunSuite with BeforeAndAfterEach with Logging {
...
     val broadcasts = allExecInfo.filter(_.exec == "BroadcastExchange")
-    assertSizeAndSupported(3, broadcasts.toSeq, Seq(Some(1154), Some(1154), Some(1855)))
+    assertSizeAndSupported(3, broadcasts.toSeq,
+      expectedDur = Seq(Some(1154), Some(1154), Some(1855)))
     val subqueryBroadcast = allExecInfo.filter(_.exec == "SubqueryBroadcast")
-    assertSizeAndSupported(1, subqueryBroadcast.toSeq, Seq(Some(1175)))
+    assertSizeAndSupported(1, subqueryBroadcast.toSeq, expectedDur = Seq(Some(1175)))
     val exchanges = allExecInfo.filter(_.exec == "Exchange")
-    assertSizeAndSupported(2, exchanges.toSeq, Seq(Some(15688), Some(8)))
+    assertSizeAndSupported(2, exchanges.toSeq, expectedDur = Seq(Some(15688), Some(8)))
```
[the remark this answers is not in the document]
Comment (nartal1) [reply tools/src/test/scala/com/nvidia/spark/rapids/tool/planparser/SqlPlanParserSuite.scala]: Updated the speedFactor for Exchange. Verified that we have tests for all the speedup factors. Had to update `fullExecName` in SubqueryBroadcastExecParser.

**Comment by nartal1** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Pull request merged**